js/base/Exchange.js

```diff
@@ -2165,7 +2172,18 @@ module.exports = class Exchange {
         };
         if (marketId !== undefined) {
             if ((this.markets_by_id !== undefined) && (marketId in this.markets_by_id)) {
-                market = this.markets_by_id[marketId];
+                const markets = this.markets_by_id[marketId];
+                const length = markets.length;
+                if (length === 1) {
+                    return markets[0];
+                } else {
+                    for (let i = 0; i < markets.length; i++) {
+                        const market = markets[i];
+                        if (market[marketType]) {
+                            return market;
+                        }
+                    }
+                }
             } else if (delimiter !== undefined) {
                 const parts = marketId.split (delimiter);
```

Comment:
Would we also have to edit `market (symbol)` to `market (symbol, marketType = undefined)` ?

Reply:
no because that takes a unified symbol and outputs a market, this PR only addresses conflicts with market ids

Reply:
But can't it accept also a marketId?

Reply:
we should default to spot if multiple markets with the same id exist

Reply:
i added and tested the edit for this.markets